functions/grapher/[slug].ts

```diff
@@ -1,33 +1,57 @@
+import { Env } from "./thumbnail/[slug].js"
+import {
+    fetchGrapherConfig,
+    getOptionalRedirectForSlug,
+    createRedirectResponse,
+} from "../_common/grapherRenderer.js"
+import { IRequestStrict, Router, error, cors } from "itty-router"
+
+const { preflight, corsify } = cors({
+    allowMethods: "GET",
```

Comment:
Ah interesting! I wasn't aware of the _headers file and was wondering if it wouldn't cause problems that we aren't doing this so far. The main point of doing cors handling like is then to make sure that an OPTIONS http request is answered, right? `Access-Control-Allow-Origin: *` is the default for the cors handler of itty router - do you think it's nicer to add it here explicitly?

Reply:
Ah, the headers file will take care of that, too.
I just checked, and the CORS rule is also present for both `HEAD` and `OPTIONS` requests.
So, maybe we should fully rely on `_headers` and put any rules in there, to avoid further confusion?

Reply:
Okay, I've now checked this:
The easiest and best way to send along a proper CORS preflight response is to add the following lines:
```ts
export const onRequestOptions: PagesFunction = async (context) => {
    const { request } = context
    return router.fetch(request)
}
```

I've also checked the itty-router source, and it does return early for an OPTIONS request - just sending along the headers and not running any application logic.

Reply:
Thanks a lot for checking, @marcelgerber! In the next PR I had added an onRequestOptions handler but not as nicely done as this. Does your solution then require the cors configuration that is present in this PR (and that I had removed in the next PR)? Or is the _headers file and the onRequestOptions enough?

Reply:
No, it needs the router instructions that you have in this PR.

It is for sure a bit weird that there is a split between the `_headers` file and the actual Functions logic. But, seeing that we can't completely support CORS with _just_ the `_headers` file, it makes sense to me to use the router's functionality for that, then. I hope that is clear enough to future-us :)